web/stripe-template/web/providers/UserProvider.jsx

```diff
@@ -0,0 +1,43 @@
+import { useFindFirst } from "@gadgetinc/react";
+import { createContext } from "react";
+import { useNavigate } from "react-router-dom";
+import { api } from "../api";
+import { useEffect } from "react";
+
+export const UserContext = createContext({});
+
+/**
+ * @param { children: import("react").ReactNode } props The props passed to the React functional component
+ *
+ * @returns { import("react").ReactElement } A React functional component
+ */
+export default ({ children }) => {
+  const navigate = useNavigate();
+
+  const [{ data: user, fetching, error }] = useFindFirst(api.user, {
+    live: true,
+    select: {
+      stripeCustomerId: true,
+      id: true,
+      priceId: true,
+    },
+  });
+
+  useEffect(() => {
+    if (!user?.stripeCustomerId && !fetching) navigate("/billing");
+  }, [user, fetching]);
+
+  if (fetching) {
+    return <div>Loading...</div>;
```

Comment:
I think this is steamrolling the entire page which is not ideal IMO

https://github.com/user-attachments/assets/451aa0e2-3d01-4dd1-a701-ab03ddce9244

Reply:
When does this happen specifically?

Reply:
Fixed it with a `userNonce` by using useUser as a way of checking if there's a user on the session. useUser's live option doesn't work properly so I need to use the useFindFirst way